pages/pools/index.tsx

```diff
@@ -55,63 +58,72 @@ const Pools: FC = () => {
 		};
 	}, [isPageOne, poolsQuery]);
 
-	const pools = useMemo(() => (poolsQuery?.data ?? []).map(parsePool), [poolsQuery?.data]);
+	const sponsors = useMemo(
+		() =>
+			(poolsQuery?.data ?? [])
+				.filter(({ id }) => !filterList.includes(id))
+				.map(({ sponsor }) => sponsor),
+		[poolsQuery?.data]
+	);
 
-	const data = useMemo(() => {
-		let list = pools
-			.filter(({ id }) => !filterList.includes(id))
-			.map(
-				({
-					sponsorFee,
-					duration,
-					sponsor,
-					name,
-					id,
-					purchaseToken,
-					contributions,
-					purchaseTokenCap,
-					purchaseTokenDecimals,
-					timestamp,
-					purchaseExpiry,
-					poolStatus,
-					hasAllowList,
-				}) => ({
-					sponsor,
-					name,
-					id,
-					purchaseToken,
-					contributions,
-					cap: purchaseTokenCap,
-					purchaseTokenDecimals,
-					duration,
-					fee: sponsorFee,
-					purchaseExpiry,
-					timestamp,
-					poolStatus,
-					hasAllowList,
-				})
-			);
+	const ensOrAddresses = useAddressesToEns(sponsors);
 
-		if (sponsorFilter != null) {
-			list = list.filter(({ sponsor }) =>
-				sponsor.toLowerCase().includes(sponsorFilter.toLowerCase())
-			);
-		}
-		if (currencyFilter != null) {
-			list = list.filter(({ purchaseToken }) =>
-				purchaseToken.toLowerCase().includes(currencyFilter.toLowerCase())
-			);
-		}
-		if (nameFilter != null) {
-			list = list.filter(({ name }) => name.toLowerCase().includes(nameFilter.toLowerCase()));
-		}
-		if (statusFilter != null) {
-			list = list.filter(({ poolStatus }) =>
-				poolStatus.toLowerCase().includes(statusFilter.toLowerCase())
-			);
-		}
-		return list;
-	}, [pools, sponsorFilter, currencyFilter, nameFilter, statusFilter]);
+	useEffect(() => {
+		const applyFilters = async () => {
+			let list = (poolsQuery?.data ?? [])
+				.filter(({ id }) => !filterList.includes(id))
+				.map(({ sponsorFee, purchaseTokenCap, ...pool }) => {
+					const parsedPool = parsePool({
+						sponsorFee,
+						purchaseTokenCap,
+						...pool,
+					});
+
+					return {
+						...parsedPool,
+						cap: purchaseTokenCap,
+						fee: sponsorFee,
+					};
+				});
+
+			if (sponsorFilter.length) {
+				list = list.filter(
+					(_, index) =>
+						(!!ensOrAddresses.length &&
+							ensOrAddresses[index].toLowerCase().includes(sponsorFilter.toLowerCase())) ||
+						sponsors[index].toLowerCase().includes(sponsorFilter.toLowerCase())
+				);
+			}
+
+			if (currencyFilter.length) {
+				list = list.filter(({ purchaseToken }) =>
+					purchaseToken.toLowerCase().includes(currencyFilter.toLowerCase())
+				);
+			}
+
+			if (nameFilter.length) {
+				list = list.filter(({ name }) => name.toLowerCase().includes(nameFilter.toLowerCase()));
+			}
+
+			if (statusFilter != null) {
+				list = list.filter(({ poolStatus }) =>
+					poolStatus.toLowerCase().includes(statusFilter.toLowerCase())
+				);
+			}
+
+			setPoolList(list);
```

Comment:
I reckon you can probably put this logic into a `useMemo` and not a `useEffect` ?

Reply:
Now, it uses an `useMemo(() => {...})` function